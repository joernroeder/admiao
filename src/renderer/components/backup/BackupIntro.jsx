import React from 'react'

import GridWrap from '../GridWrap'
import Cell from '../Cell'
import Heading from '../Heading'
import Text from '../Text'
import Button from '../Button'

const BackupIntro = () => {
	return (
		<>
			<GridWrap
				flexDirection={'column'}
				justifyContent={'_space-around'}
				height={'100vh'}
			>
				<Cell>
					<GridWrap>
						<Cell gridColumn={8} gridOffset={1}>
							<Heading mt={[4, 6, 8]}>Backup Seed</Heading>
						</Cell>
					</GridWrap>
					<GridWrap>
						<Cell gridColumn={[10, 8, 5]} gridOffset={2} mt={4}>
							<Text>
								Donec ullamcorper nulla non metus auctor
								fringilla. Duis mollis, est non commodo luctus,
								nisi erat porttitor ligula, eget lacinia odio
								sem nec elit.
							</Text>
						</Cell>
					</GridWrap>
				</Cell>
				<Cell>
					<GridWrap justifyContent={'flex-end'} mt={9}>
						<Cell>
							<Button to={'./select-parts'} mb={[4, 6, 8]}>
								Start Backup
							</Button>
						</Cell>
						<Cell gridColumn={[1]} />
					</GridWrap>
				</Cell>
			</GridWrap>
		</>
	)
}
export default BackupIntro
