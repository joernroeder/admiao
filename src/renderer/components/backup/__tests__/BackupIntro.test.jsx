import React from 'react'

import { fireEvent, render } from '@testing-library/react'

import BackupIntro from '../BackupIntro'

describe('BackupIntro view', () => {
	test('to have headline, description and button', () => {
		const { getByText, container, debug } = render(<BackupIntro />)

		// headline
		expect(getByText('Backup Seed')).toBeInTheDocument()

		// description
		expect(getByText(/Donec/i, { exact: false })).toBeInTheDocument()

		// next button
		expect(getByText('Start Backup')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	test('start button should navigate to the part selection at "//select-parts"', () => {
		const { history, getByText } = global.renderWithRouter(<BackupIntro />)

		fireEvent.click(getByText('Start Backup'))

		expect(history.location.pathname).toEqual('//select-parts')
	})
})
