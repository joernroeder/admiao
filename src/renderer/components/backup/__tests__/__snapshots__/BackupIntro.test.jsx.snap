// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackupIntro view to have headline, description and button 1`] = `
.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 100vh;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 4px;
  padding-right: 4px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 4px;
  padding-right: 4px;
  width: 66.66666666666666%;
}

.emotion-0 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  line-height: 2;
  -webkit-transform: 2px;
  -ms-transform: 2px;
  transform: 2px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    margin-top: 128px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    margin-top: 512px;
  }
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 20px;
    line-height: 3;
    -webkit-transform: 3px;
    -ms-transform: 3px;
    transform: 3px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 32px;
  padding-left: 4px;
  padding-right: 4px;
  width: 83.33333333333334%;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    width: 66.66666666666666%;
  }
}

@media screen and (min-width:52em) {
  .emotion-4 {
    width: 41.66666666666667%;
  }
}

.emotion-3 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 9px;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  margin-bottom: 32px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: white;
  background-color: black;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

@media screen and (min-width:40em) {
  .emotion-8 {
    margin-bottom: 128px;
  }
}

@media screen and (min-width:52em) {
  .emotion-8 {
    margin-bottom: 512px;
  }
}

.emotion-8 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-8:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-8:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-8:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-8:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-8 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-7 {
  background-color: whiteTransparent;
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-7,
*:focus > .emotion-7 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 4px;
  padding-right: 4px;
  width: 8.333333333333332%;
}

<div
  class="emotion-13"
>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-1"
      >
        <h1
          class="emotion-0"
          color="black"
          font-size="2,2,3"
        >
          Backup Seed
        </h1>
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-4"
      >
        <p
          class="emotion-3"
          color="black"
          font-size="0"
        >
          Donec ullamcorper nulla non metus auctor fringilla. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit.
        </p>
      </div>
    </div>
  </div>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-11"
    >
      <div
        class="emotion-6"
      >
        <a
          class="emotion-8"
          href="//select-parts"
        >
          <span>
            Start Backup
             
            <span
              class="icon next"
            >
              →
            </span>
          </span>
          <span
            class="emotion-7"
          />
        </a>
      </div>
      <div
        class="emotion-10"
      />
    </div>
  </div>
</div>
`;
